registry/Scripts/pre-reqs.ps1

```diff
@@ -1,82 +1,642 @@
+
+.Parameter RegistryUserPassword
+    Password using which images will be push and pull.
+
+.Example
+  Set-ContainerRegistryPrerequisite
+
+#>
+function Set-ContainerRegistryPrerequisite
+(
+    [Parameter(Mandatory = $true, HelpMessage = "Location of Azure Stack.")]
+    [string] $Location,
+    [Parameter(Mandatory = $true, HelpMessage = "Service principle id which will be added to provide contributor access.")]
+    [string] $ServicePrincipleId,
+    [Parameter(Mandatory = $true, HelpMessage = "Name of the resource group to be created.")]
+    [string] $ResourceGroupName,
+    [Parameter(Mandatory = $true, HelpMessage = "Storage account name which needs to be created.")]
+    [string] $StorageAccountName,
+    [Parameter(Mandatory = $true, HelpMessage = "Storage account blob container name which needs to be created.")]
+    [string] $StorageAccountBlobContainer,
+    [Parameter(Mandatory = $true, HelpMessage = "Keyvault name which needs to be created.")]
+    [string] $KeyVaultName,    
+    [Parameter(Mandatory = $true, HelpMessage = "Secret name for the certificate which needs to be created in keyvault.")]
+    [string] $CertificateSecretName,
+    [Parameter(Mandatory = $true, HelpMessage = "Full file path where certificate file is present.")]
+    [string] $CertificateFilePath,
+    [Parameter(Mandatory = $true, HelpMessage = "Password to read the certificate.")]
+    [string] $CertificatePassword, 
+    [Parameter(Mandatory = $true, HelpMessage = "User name using which images will be push and pull.")]
+    [string] $RegistryUserName,
+    [Parameter(Mandatory = $true, HelpMessage = "Password using which images will be push and pull.")]
+    [string] $RegistryUserPassword
+)
+{
+    # Assuming tenant is logged in already and selected the given subscription.
+    # best case check if tenant is loged in. Currently subscription selected status is not checked.
+    Get-AzureStackLoginStatus
+
+    # Create resource group. In case exist skip creation.
+    New-ResourceGroup -ResourceGroupName $ResourceGroupName
+
+    # Create storage account. In case exist skip creation.
+    $storageAccountDetails = New-StorageAccount -ResourceGroupName $ResourceGroupName `
+                                    -Location $Location `
+                                    -StorageAccountName $StorageAccountName
+
+    # Create current context to given storage account.
+    Set-AzureRmCurrentStorageAccount -ResourceGroupName $ResourceGroupName `
+                                     -AccountName $StorageAccountName | out-null
+
+    # Create new storage blob container.
+    New-StorageAccountContainer -ResourceGroupName $ResourceGroupName `
+                                -StorageAccountName $StorageAccountName `
+                                -StorageAccountBlobContainer $StorageAccountBlobContainer
+
+    Write-Host "Check if storage account($StorageAccountName) already gave access to ApplicationId($ServicePrincipleId)"
```

Comment:
sa does not grant access: "checking if client has access to sa"
account_($StorageAccountName)